src/evo/core_write.cpp

```diff
@@ -74,7 +75,7 @@
     if (IsServiceDeprecatedRPCEnabled()) {
         ret.pushKV("service", netInfo->GetPrimary().ToStringAddrPort());
     }
-    ret.pushKV("addresses", netInfo->ToJson());
+    ret.pushKV("addresses", ShimNetInfoPlatform(*this, nType));
```

Comment:
let's rename this, to make it clearer how it is intended to be used. This is a bad name, but `MakeNetInfoFromNetInfoOrLegacy` or something like that. @coderabbitai any ideas on a better name?

Reply:
`ShimNetInfoPlatform()` has been renamed to `GetNetInfoWithLegacyFields()` and `ShimPlatformPort()` has been renamed to `GetPlatformPort()`, resolved in latest push.